### PR TITLE
test(datafusion): cover SQL schema evolution reads

### PR DESCRIPTION
**Pull request by liujiwen-up**
### Purpose

Linked issue:  #258

Add DataFusion SQL coverage for schema evolution tables generated from Java/Spark consistency fixtures, focusing on mixed-format reads through SQL. The rename coverage uses `schema_evolution_rename_column`, whose fixture contains parquet/orc/avro files.

### Brief change log

- Add a small `assert_sql` helper for table-formatted SQL result assertions.
- Cover SQL reads for mixed-format add column and type promotion tables.
- Cover SQL reads for rename, drop, and reorder/move schema evolution cases with `SELECT`, `WHERE`, and `ORDER BY id`.

### Tests

- `cargo fmt --check`: passed
- `git diff --check`: passed
- `cargo test -p paimon-datafusion --test read_tables schema_evolution --no-run`: passed
- `cargo test -p paimon-datafusion --test read_tables schema_evolution -- --nocapture`: not fully runnable locally because `/tmp/paimon-warehouse` has not been provisioned and Docker daemon is unavailable

### API and Format

No API or storage format changes.

### Documentation

No documentation changes required.

### CODE REVIEW

File: crates/integrations/datafusion/tests/read_tables.rs [hunk 1]
```diff
@@ -98,6 +99,15 @@ async fn collect_query(
     ctx.sql(sql).await?.collect().await
 }
 
+async fn assert_sql(sql: &str, expected: &[&str]) {
```
Comment (QuakeWang) [inline crates/integrations/datafusion/tests/read_tables.rs]: Nit: this helper compares the full `pretty_format_batches` output verbatim. It keeps the expected SQL result readable, but also couples the tests to Arrow/DataFusion pretty-printer details such as padding, borders, and NULL rendering. Fine for these small result sets; if we want less brittle tests across dependency upgrades, comparing normalized rows/columns would be more stable.

Comment (liujiwen-up) [reply crates/integrations/datafusion/tests/read_tables.rs]: thanks,i done

**Review by QuakeWang — APPROVED** [review]
+1
